location/src/main/java/org/odk/collect/location/satellites/GpsStatusSatelliteInfoClient.kt

```diff
@@ -0,0 +1,37 @@
+package org.odk.collect.location
+
+import android.annotation.SuppressLint
+import android.location.GpsStatus
+import android.location.LocationManager
+import org.odk.collect.androidshared.livedata.NonNullLiveData
+import org.odk.collect.location.satellites.SatelliteInfoClient
+
+class GpsStatusSatelliteInfoClient(private val locationManager: LocationManager) :
+    SatelliteInfoClient {
+
+    override val satellitesUsedInLastFix: NonNullLiveData<Int>
+        get() = GpsStatusSatellitesLiveData(locationManager)
+}
+
+@SuppressLint("MissingPermission")
+@Suppress("deprecation")
```

Comment:
The deprecated API has been replaced by a new one that is only available in API 24 and up. We probably want to come back here and either use `LocationManagerCompat` or just switch between two different `LiveData` implementations based on OS. I'm thinking that can just be an issue we schedule for the next release rather than blocking this one?